src/renderer/coremods/settings/pages/General.tsx

```diff
@@ -86,42 +80,84 @@ export const General = (): React.ReactElement => {
-            })
-            .then((answer) => {
-              if (answer) {
-                expOnChange(value);
-                setTimeout(() => window.location.reload(), 250);
-              }
-            });
-        }}
-        note={Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS_DESC.format()}>
-        {Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS}
-      </SwitchItem>
-
       <SwitchItem
         {...util.useSetting(generalSettings, "autoApplyQuickCss")}
         note={Messages.REPLUGGED_SETTINGS_QUICKCSS_AUTO_APPLY_DESC}>
         {Messages.REPLUGGED_SETTINGS_QUICKCSS_AUTO_APPLY}
       </SwitchItem>
 
-      <ButtonItem
-        button={Messages.REPLUGGED_SETTINGS_DEV_COMPANION_RECONNECT}
-        note={Messages.REPLUGGED_SETTINGS_DEV_COMPANION_DESC}
-        onClick={() => {
-          socket?.close(1000, "Reconnecting");
-          initWs(true);
-        }}>
-        {Messages.REPLUGGED_SETTINGS_DEV_COMPANION}
-      </ButtonItem>
+      <Category
+        title={Messages.REPLUGGED_SETTINGS_ADVANCED}
+        note={Messages.REPLUGGED_SETTINGS_ADVANCED_DESC}>
+        <FormItem
+          title={Messages.REPLUGGED_SETTINGS_BACKEND}
+          note={Messages.REPLUGGED_SETTINGS_BACKEND_DESC}
+          divider={true}
+          style={{ marginBottom: "20px" }}>
+          <TextInput
+            {...util.useSetting(generalSettings, "apiUrl")}
+            placeholder="https://replugged.dev"
+          />
+        </FormItem>
+
+        <SwitchItem
+          value={expValue}
+          onChange={(value) => {
+            void modal
+              .confirm({
+                title: Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS,
+                body: Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS_DESC.format(),
+                confirmText: Messages.REPLUGGED_RELOAD,
+                confirmColor: Button.Colors.RED,
+              })
+              .then((answer) => {
+                if (answer) {
+                  expOnChange(value);
+                  setTimeout(() => window.location.reload(), 250);
+                }
+              });
+          }}
+          note={Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS_DESC.format()}>
+          {Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS}
+        </SwitchItem>
+
+        <SwitchItem
+          value={rdtValue}
+          onChange={(value) => {
+            void modal
+              .confirm({
+                title: Messages.REPLUGGED_SETTINGS_REACT_DEVTOOLS,
+                body: Messages.REPLUGGED_SETTINGS_REACT_DEVTOOLS_DESC.format(),
+                confirmText: Messages.REPLUGGED_RELOAD,
+                confirmColor: Button.Colors.RED,
+              })
+              .then((answer) => {
+                if (answer) {
+                  if (value) {
+                    void RepluggedNative.reactDevTools.downloadExtension().then(() => {
+                      rdtOnChange(value);
+                      setTimeout(() => window.DiscordNative.app.relaunch(), 250);
```

Comment:
Let's create functions here called reload and relaunch (with the timeout built in) and then replace the calls in this file.